client/signup/steps/difm-site-picker/index.tsx

```diff
@@ -0,0 +1,196 @@
+import { Button, Card, Dialog } from '@automattic/components';
+import { useTranslate } from 'i18n-calypso';
+import { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import FormLabel from 'calypso/components/forms/form-label';
+import FormTextInput from 'calypso/components/forms/form-text-input';
+import SiteSelector from 'calypso/components/site-selector';
+import StepWrapper from 'calypso/signup/step-wrapper';
+import { saveSignupStep, submitSignupStep } from 'calypso/state/signup/progress/actions';
+import { getSiteDomain, getSiteSlug, getSiteTitle } from 'calypso/state/sites/selectors';
+import { SiteData } from 'calypso/state/ui/selectors/get-selected-site';
+
+interface Props {
+	stepSectionName: string | null;
+	stepName: string;
+	goToStep: () => void;
+	goToNextStep: () => void;
+}
+
+const DIFMSitePicker = ( {
+	filter,
+	onSiteSelect,
+}: {
+	filter: ( site: SiteData ) => boolean;
+	onSiteSelect: ( siteId: number ) => void;
+} ) => {
+	return (
+		<Card className="difm-site-picker__wrapper">
+			<SiteSelector filter={ filter } onSiteSelect={ onSiteSelect } />
+		</Card>
+	);
+};
+
+export default function DIFMSitePickerStep( props: Props ): React.ReactElement {
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+	const { goToNextStep } = props;
+	const [ siteId, setSiteId ] = useState< number | null >( null );
+	const [ confirmDomain, setConfirmDomain ] = useState( '' );
+	const siteDomain = useSelector( ( state ) => getSiteDomain( state, siteId ) );
+	const siteSlug = useSelector( ( state ) => getSiteSlug( state, siteId ) );
+	const siteTitle = useSelector( ( state ) => getSiteTitle( state, siteId ) );
+	const headerText = translate( 'Choose where you want us to build your site.' );
+	const subHeaderText = translate( 'Some unsupported sites may be hidden.' );
+
+	useEffect( () => {
+		dispatch( saveSignupStep( { stepName: props.stepName } ) );
+	}, [ dispatch, props.stepName ] );
+
+	const handleSiteSelect = ( siteId: number ) => {
+		setSiteId( siteId );
+	};
+
+	const filterSites = ( site: SiteData ) => {
+		return (
+			site.capabilities?.manage_options &&
+			! site.jetpack &&
+			! site.options?.is_wpforteams_site &&
```

Comment:
These two lines are also TS errors which you can fix by adding `is_wpforteams_site: boolean` and `is_difm_lite_in_progress: boolean` to [SiteDataOptions](https://github.com/Automattic/wp-calypso/blob/0b4f87169f148a87d4e44a4f868921bc74ab08cc/client/state/ui/selectors/site-data.ts#L26).